src/lwc/geGiftEntryFormApp/geGiftEntryFormApp.js

```diff
@@ -289,55 +293,72 @@ export default class GeGiftEntryFormApp extends NavigationMixin(LightningElement
                     await this.processBatch();
                 }
             }
+            return;
+        }
+        this.handleBatchProcessingErrors();
+    }
+
+    handleBatchProcessingErrors() {
+        if (this.expectedCountOfGifts && this.expectedTotalBatchAmount) {
+            handleError(geBatchGiftsExpectedTotalsMessage);
         } else {
-            if (this.expectedCountOfGifts && this.expectedTotalBatchAmount) {
-                handleError(geBatchGiftsExpectedTotalsMessage);
-            } else {
-                handleError(geBatchGiftsExpectedCountOrTotalMessage);
-            }
+            handleError(geBatchGiftsExpectedCountOrTotalMessage);
         }
     }
 
-    async processBatch() {
-        await this.processPayments();
-        this.navigateToDataImportProcessingPage(); 
+    get isBatchProcessing() {
+        return this._isBatchProcessing;
     }
 
-    async processPayments() {
-        await processPayments({
-                batchId: this.batchId
+    async processBatch() {
+        await processBatch({
+            batchId: this.batchId
         }).catch(error => {
             handleError(error);
+        }).finally(() => {
+            this.startPolling();
         });
     }
 
-    navigateToDataImportProcessingPage() {
-        let url = '/apex/' + this.bdiDataImportPageName() +
-            '?batchId=' + this.recordId + '&retURL=' + this.recordId;
-
-        this[NavigationMixin.Navigate]({
-                type: 'standard__webPage',
-                attributes: {
-                    url: url
+    async startPolling() {
+        const poll = window.setInterval(() => {
+            Promise.resolve(true).then(() => {
+                this.refreshBatchTotals();
+            }).then(() => {
+                if (!this._isBatchProcessing) {
+                    this.handleProcessedBatch();
+                    window.clearInterval(poll);
                 }
-            },
-            true
+            })
+        }, 5000);
+    }
+
+    handleProcessedBatch() {
+        this.collapseForm();
+        showToast(
+            this.CUSTOM_LABELS.PageMessagesConfirm,
+            GeLabelService.format(
+                this.CUSTOM_LABELS.geBatchProcessingSuccess,
+                [this.batchName]),
+            'success',
+            'dismissible',
+            null
         );
     }
 
+    collapseForm() {
+        const form = this.template.querySelector('c-ge-form-renderer');
+        form.collapse();
+    }
+
     async refreshBatchTotals() {
-        if(this.isElevateCustomer) {
+        if (this.isElevateCustomer) {
```

Comment:
With this change to move the batch totals outside of this condition - which totally makes sense - I think we might drop this condition entirely? setting `_hasDisplayedExpiredAuthorizationWarning `to false shouldn't really have any impact to non-elevate customers and I think we can simplify by avoiding the` if (this.isElevateCustomer)` call entirely?